app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/uploadsupportingdocs/UploadSupportingEvidenceController.scala

```diff
@@ -0,0 +1,1012 @@
+                ) =>
+              CompleteUploadSupportingEvidenceAnswers(
+                doYouWantToUploadSupportingEvidenceAnswer,
+                supportingEvidences.filterNot(c => c.reference == id)
+              )
+          }
+          newDraftReturn = fillingOutReturn.draftReturn match {
+            case s @ DraftSingleDisposalReturn(
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _
+                ) =>
+              s.copy(uploadSupportingDocuments = Some(updatedAnswers))
+            case m @ DraftMultipleDisposalsReturn(
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _
+                ) =>
+              m.copy(uploadSupportingDocuments = Some(updatedAnswers))
+          }
+
+          _ <- returnsService
+                .storeDraftReturn(
+                  newDraftReturn,
+                  fillingOutReturn.subscribedDetails.cgtReference,
+                  fillingOutReturn.agentReferenceNumber
+                )
+          _ <- EitherT(
+                updateSession(sessionStore, request)(
+                  _.copy(journeyStatus = Some(fillingOutReturn.copy(draftReturn = newDraftReturn)))
+                )
+              )
+        } yield ()
+
+        result.fold({ e =>
+          logger.warn("Could not update session", e)
+          errorHandler.errorResult()
+        }, _ => Redirect(routes.UploadSupportingEvidenceController.checkYourAnswers()))
+      }
+  }
+
+  def checkYourAnswers(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    withUploadSupportingEvidenceAnswers(request) { (_, _, _, fillingOutReturn, answers) =>
+      (answers, fillingOutReturn.draftReturn) match {
+        case (c: UploadSupportingEvidenceAnswers, s: DraftSingleDisposalReturn) => {
+          checkYourAnswersHandler(c, fillingOutReturn, s)
+        }
+        case (c: UploadSupportingEvidenceAnswers, s: DraftMultipleDisposalsReturn) =>
+          checkYourAnswersHandler(c, fillingOutReturn, s)
+        case _ =>
+          logger.warn("error processing check your answers")
+          errorHandler.errorResult()
+      }
+    }
+  }
+
+  def checkYourAnswersSubmit(): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withUploadSupportingEvidenceAnswers(request) { (_, _, _, fillingOutReturn, answers) =>
+        val updatedAnswers: UploadSupportingEvidenceAnswers = answers match {
+          case IncompleteUploadSupportingEvidenceAnswers(None, _) =>
+            CompleteUploadSupportingEvidenceAnswers(
+              doYouWantToUploadSupportingEvidence = false,
+              List.empty
+            ) // shut the compiler up - this can never happen actually
```

Comment:
if you don't think this is going to happen then it's better to show an error page, just in case it actually does

Reply:
will be addressed after 6th April